temporalio/common.py

```diff
@@ -1230,3 +1230,9 @@ def _type_hints_from_func(
         # necessarily
         args.append(arg_hint)  # type: ignore
     return args, ret
+
+
+class WorkflowLogicFlag(IntEnum):
+    """Flags that may be set on task/activation completion to differentiate new from old workflow behavior."""
+
+    RAISE_ON_CANCELLING_COMPLETED_ACTIVITY = 1
```

Comment:
Even though we only think it might be on activities and we struggle to confirm behavior on child workflows, how confident are you that this behavior is not a problem with child workflows? I guess the question is which is a better approach "untestable 'fix' for this for child workflows" or "wait until it may come up to add a separate flag for child workflows at that time"? I am on the fence myself.